magma/bits.py

```diff
@@ -72,13 +75,94 @@ class BitsMeta(AbstractBitVectorMeta, ArrayMeta):
     def __new__(mcs, name, bases, namespace, info=(None, None, None), **kwargs):
         return ArrayMeta.__new__(mcs, name, bases, namespace, info, **kwargs)
 
+    def _make_const(cls, value):
```

Comment:
It seems a bit odd to make them free if they take `cls` and now we have to pass it explicitly? I'd normally be fine with free functions when `cls` is not used, but the implicit `cls` argument seems warranted here since the logic uses the reference.